swap/int256/int256.go

```diff
@@ -30,6 +30,11 @@ type Int256 struct {
 	value *big.Int
 }
 
+// BigIntWrapper represents a struct with an underlying big.Int value
+type BigIntWrapper interface {
+	Value() *big.Int
```

Comment:
why is this needed?

Reply:
i implemented this interface to be able to compare (`Cmp` and `Equals`) `Int256`- and `Uint256`-typed variables.

it is not _strictly_ needed, but it is (i think) tidier than calling `Value` on each of these differently-typed variables and then calling the underlying `Cmp` func on them, specially when we _already_ have this call on each of these types.

Reply:
but in the case that you need to compare both, you can just call `Value()` on the `Int256` and `Uint256` and use the `big.Int` comparators right? not sure if another type is needed here for these edge cases.

Reply:
unless, you'll need to do these sort of comparisons all over the place, but then it puts in question the necessity of the `Int256` and `Uint256` types in the first place

Reply:
it's not all over the place, but it does occur in a couple of instances as far as i recall.

i have to ask: why would it be preferable to call `Value` on 2 variables, then call `Cmp`, when i can do it in just 1 call? it also makes sense from a logical point of view, you should always be able to tell whether a number is bigger, equal or smaller than another.

the necessity of the `Int256` and `Uint256` exists because of the smart contracts. we want the corresponding variables on the go side to have the same range as their SC counterparts. that's the reason for all these PRs.